ouroboros-consensus/src/Ouroboros/Storage/ChainDB/API.hs

```diff
@@ -286,6 +309,63 @@ data IteratorResult blk =
     --
     -- This will only happen when streaming very old forks.
 
+data UnknownRange blk =
+    -- | The block at the given point was not found in the ChainDB.
+    MissingBlock (Point blk)
+    -- | The requested range forks off too far in the past, i.e. it doesn't
+    -- fit on the tip of the ImmutableDB.
+  | ForkTooOld (StreamFrom blk)
+  deriving (Eq, Show)
+
+-- | Check whether the bounds make sense
+--
+-- An example of bounds that don't make sense:
+--
+-- > StreamFromExclusive (Point { pointSlot = SlotNo 3 , .. }
+-- > StreamToInclusive   (Point { pointSlot = SlotNo 3 , .. }
+validBounds :: StreamFrom blk -> StreamTo blk -> Bool
+validBounds from to = case from of
+  StreamFromInclusive (Point { pointHash = GenesisHash }) -> False
+
+  StreamFromInclusive (Point { pointSlot = sfrom }) -> case to of
+    StreamToInclusive (Point { pointSlot = sto }) -> sfrom <= sto
+    StreamToExclusive (Point { pointSlot = sto }) -> sfrom <  sto
+
+  StreamFromExclusive (Point { pointSlot = sfrom }) -> case to of
+    StreamToInclusive (Point { pointSlot = sto }) -> sfrom <  sto
+    StreamToExclusive (Point { pointSlot = sto }) -> sfrom <  sto
+
+-- | Stream all blocks from the current chain.
+--
+-- To stream all blocks from the current chain from the ChainDB, one would use
+-- @'StreamFromExclusive' 'genesisPoint'@ as the lower bound and
+-- @'StreamToInclusive' tip@ as the upper bound where @tip@ is retrieved with
+-- 'getTipPoint'.
+--
+-- However, when the ChainDB is empty, @tip@ will be 'genesisPoint' too, in
+-- which case the bounds don't make sense. This function correctly handles
+-- this case.
+--
+-- Note that this is note a 'Reader', so the stream will not include blocks
```

Comment:
note/not

Good clarification on the difference between a reader and an iterator: one very much live, the other as static as practically possible. And true, they could be confused. Perhaps we should put this note in the docs for the iterator and/or reader types.